src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -678,8 +687,31 @@ public void Commit(bool previewChanges = false)
         /// <returns><see langword="true"/> if the rename operation was commited, <see
         /// langword="false"/> otherwise</returns>
         private bool CommitWorker(bool previewChanges)
+        {
+            return _threadingContext.JoinableTaskFactory.Run(() => CommitWorkerAsync(previewChanges));
+        }
+
+        public Task CommitAsync(bool previewChanges = false)
+            => CommitWorkerAsync(previewChanges);
+
+        private async Task<bool> CommitWorkerAsync(bool previewChanges)
         {
             _threadingContext.ThrowIfNotOnUIThread();
+
+            var indicatorFactory = _workspace.Services.GetRequiredService<IBackgroundWorkIndicatorFactory>();
+
+            // We do not cancel on edit because as part of the rename system we have asynchronous work still occurring
+            // that itself may be asynchronously editing the buffer (for example, updating reference locations with the
+            // final renamed text).  Ideally though, once we start comitting, we would cancel any of that work and then
+            // only have the work of rolling back to the original state of the world and applying the desired edits
+            // ourselves.
+            using var context = indicatorFactory.Create(
```

Comment:
Definitely open to trying this out, I just have a feeling I might accidentally cancel, or might be not as clear when its done

Reply:
definitely possible :)

Reply:
tracking this concern with: https://github.com/dotnet/roslyn/issues/61416